test/widgets/test_window_count.py

```diff
@@ -1,70 +1,38 @@
-from libqtile import hook
-from libqtile.bar import Bar
+import libqtile
 from libqtile.widget import WindowCount
 
 
-class FakeWindowList:
-    def __init__(self):
-        self.windows = []
+def test_window_count(manager_nospawn):
+    class WindowCountConf(libqtile.confreader.Config):
+        groups = [libqtile.config.Group("a"), libqtile.config.Group("b")]
+        screens = [
+            libqtile.config.Screen(
+                top=libqtile.bar.Bar([WindowCount()], 10)
+            )
+        ]
 
-
-class FakeQtile:
-    def __init__(self):
-        self.groups = {1: FakeWindowList(), 0: FakeWindowList()}
-        self._current_group = 1
-
-    def add_window(self):
-        self.groups[self._current_group].windows.append("window")
-        hook.fire("client_managed", "window")
-
-    def close_window(self):
-        try:
-            self.groups[self._current_group].windows.pop()
-            hook.fire("client_killed", "window")
-        except IndexError:
-            pass
-
-    def switch_group(self):
-        self._current_group = (self._current_group + 1) % 2
-        hook.fire("setgroup")
-
-    def draw(self):
-        pass
-
-    @property
-    def current_group(self):
-        return self.groups[self._current_group]
-
-
-def test_window_count():
-    wc = WindowCount()
-    fakebar = Bar([wc], 24)
-    wc.bar = fakebar
-    qtile = FakeQtile()
-    wc.qtile = qtile
-    wc._setup_hooks()
+    manager_nospawn.start(WindowCountConf)
 
     # No windows opened
-    assert wc._count == 0
+    assert int(manager_nospawn.c.widget["windowcount"].get()) == 0
```

Comment:
do these need the `int`s?

Reply:
I think so; the return value is a string.

Reply:
Ah yep, of course.